13-Tenant Security Solution Setup/Scripts/RemediationScripts/Remediate-ConfigASCTier.ps1

```diff
@@ -0,0 +1,389 @@
+
+    # Safe Check: Checking whether the current account is of type User
+    if($currentSub.Account.Type -ne "User")
+    {
+        Write-Host "Warning: This script can only be run by user account type." -ForegroundColor Yellow
+        break;
+    }
+
+    # Safe Check: Current user need to be either Contributor or Owner for the subscription
+    $currentLoginRoleAssignments = Get-AzRoleAssignment -SignInName $currentSub.Account.Id -Scope "/subscriptions/$($SubscriptionId)";
+
+    if(($currentLoginRoleAssignments | Where { $_.RoleDefinitionName -eq "Owner"  -or $_.RoleDefinitionName -eq 'Contributor' } | Measure-Object).Count -le 0)
+    {
+        Write-Host "Warning: This script can only be run by an Owner or Contributor of subscription [$($SubscriptionId)] " -ForegroundColor Yellow
+        break;
+    }
+
+    # Declaring required ASC type and pricing tier
+    $reqASCTierResourceTypes = "VirtualMachines","SqlServers","AppServices","StorageAccounts","KubernetesService","ContainerRegistry","KeyVaults","SqlServerVirtualMachines";		
+    $reqASCTier = "Standard";
+    $reqProviderName = "Microsoft.Security"
+    $isProviderRegister = $true
+
+    # Checking IsProviderRegister with 'Microsoft.Security' provider
+    $registeredProvider =  Get-AzResourceProvider -ListAvailable | Where-Object { $_.RegistrationState -eq "Registered" -and $_.ProviderNamespace -eq $reqProviderName }
+
+    if($null -eq $registeredProvider)
+    {
+        # capture provider registration state
+        $isProviderRegister = $false
+        Write-Host "Found [$($reqProviderName)] provider is not registered."
+        Write-Host "$reqProviderName registering [It takes 2-3 min to get registered]..."
+        # Registering provider with required provider name, it will take 1-2 min for registration
+        try 
+        {
+            Register-AzResourceProvider -ProviderNamespace $reqProviderName
+            while((Get-AzResourceProvider -ListAvailable | Where-Object { $_.ProviderNamespace -eq $reqProviderName }).RegistrationState -ne "Registered")
+            {
+                Start-Sleep -Seconds 10
+                Write-Host "$reqProviderName registering..." -ForegroundColor Yellow
+            }    
+        }
+        catch 
+        {
+            Write-Host "Error Occured while registering $reqProviderName provider. ErrorMessage [$($_)]" -ForegroundColor Red
+            break
+        }
+        Write-Host "$reqProviderName provider successfully registered." -ForegroundColor Green
+    }
+
+    Write-Host "Step 2 of 3: Checking [$($reqASCTier)] pricing tier for [$($reqASCTierResourceTypes)] ASC type..."
+    $nonCompliantASCTierResourcetype = @()
+    $nonCompliantASCTierResourcetype = Get-AzSecurityPricing | Where-Object { $_.PricingTier -ne $reqASCTier -and $reqASCTierResourceTypes.Contains($_.Name) }
+
+    $nonCompliantASCTypeCount = ($nonCompliantASCTierResourcetype | Measure-Object).Count
```

Comment:
If there are no non compliant ASC type we should return or skip other steps.

Reply:
Yes, In case of 'Microsoft.Security' provider is already registered and no non-compliant ASC type found

Added condition to return